### ISSUE TITLE
Add the Selection Panel happy path view (Storybook)

### ISSUE THREAD

**Issue by techanvil** [google/site-kit-wp#8157]
## Feature Description

Create the Selection Panel component, implementing the happy path view, and add it to Storybook.

This should include the CTAs but not the behaviour for saving the selection, this will be handled separately via https://github.com/google/site-kit-wp/issues/8158.

Additionally it should not include the loading state, the Audience Creation Notice or the Selection Panel Info Notice, these will be also handled separately via https://github.com/google/site-kit-wp/issues/8162, https://github.com/google/site-kit-wp/issues/8164 and https://github.com/google/site-kit-wp/issues/8159.

It _should_ include the logic/behaviour described in the **selection panel > available audiences** section.

See [selection panel](https://docs.google.com/document/d/1MGD5Djy6AeeZC4zBtHqS-lQEWD9jw0kf-IIWw-jLCFU/edit#heading=h.qt81tly4no4r) in the design doc.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The Audience Segmentation Selection Panel component should be created and stories for it added to Storybook.
* It should follow the [Figma design](https://www.figma.com/file/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?type=design&node-id=1106-16323&mode=design&t=GPwf11sg1JqGPeDf-0), and should also be based on the Key Metrics selection panel.
* The generic components extracted from the KM Selection Panel should be reused (see google/site-kit-wp#8652), with changes to the design for Audience Segmentation applied back to the KM panel (notably the footer layout in the mobile view as mentioned in the [design doc](https://docs.google.com/document/d/1MGD5Djy6AeeZC4zBtHqS-lQEWD9jw0kf-IIWw-jLCFU/edit#heading=h.qt81tly4no4r)).
* The Selection Panel should display the list of available audiences that is cached in the DB (i.e. stored in the `availableAudiences` setting). Their display names and descriptions should be displayed directly, with the exception of the "All users" default audience which should be shown as "All visitors".
* The “Purchasers” default audience should only be listed if it’s had any users in it since the associated Analytics property was created.
* Audiences should be selectable via checkboxes and, when there is a currently saved selection, grouped into "Current selection" (the currently saved selection) and "Additional groups" groups.
* The user count for an audience should be displayed on the right hand side of it.
* The Selection Panel should display the header text including the link to the Admin Settings tab on the Settings screen.
  * If the Audience Segmentation Settings section is available at implementation time this link should ensure the section is scrolled into view when navigating to the Settings screen. Otherwise, a small followup issue should be created to update the link when the section is available.
* The bottom part of the Selection Panel should include:
  * The "Learn more" text including the Analytics link pointing to the [introduction to audiences in Google Analytics](https://support.google.com/analytics/answer/12799087) article.  
  * The footer which includes the selection count and **Cancel** / **Save selection** CTAs.
    * As alluded to in the Feature Description, the CTAs don't need to be hooked up here, this will be done via [google/site-kit-wp#8158](https://github.com/google/site-kit-wp/issues/8158).

---------------

* Note that the special case for retrieving reports using the newVsReturning dimension to avoid the “partial data” state as discussed in the design doc's **selection panel > available audiences** section should not be implemented here, this will be handled separately via [google/site-kit-wp#8160](https://github.com/google/site-kit-wp/issues/8160).
* Note too that the specific ordering of audiences within the selection panel will be handled via the subsequent issue #8519.

## Implementation Brief 

Please refer POC PR: https://github.com/google/site-kit-wp/pull/8676.
Also, this implementation will be done once #8652 is merged.

- [x] Create directory `AudiencesSelectionPanel` inside `assets/js/modules/analytics-4/components/AudienceSegmentation/`
  - [x] Create `constants.js` inside new directory. Add the following constants in it [similar to key metrics component](https://github.com/google/site-kit-wp/blob/8b33773dd1457d710d2bbcc8d2a744c502055e8b/assets/js/components/KeyMetrics/constants.js#L19-L25).
    - [x] Add `AUDIENCES_SELECTION_PANEL_OPENED_KEY`, `AUDIENCES_SELECTION_FORM`, `AUDIENCES_SELECTED`, `MIN_SELECTED_AUDIENCES_COUNT` and `MAX_SELECTED_AUDIENCES_COUNT`.
  - [x] Create `index.js` which will have `AudiencesSelectionPanel` component in it.
    - [x] It should use the `SelectionPanel` component and pass the relevant props to it.
    - [x] Make use of `AUDIENCE_SELECTION_PANEL_OPENED_KEY` constant to pass `isOpen` prop to `SelectionPanel`.
    - [x] `onOpen` prop should accept a callback function where the callback should add the list of saved audiences to `AUDIENCES_SELECTED`. List of saved audiences can be retrieved using `getConfiguredAudiences` [selector](https://github.com/google/site-kit-wp/blob/8b33773dd1457d710d2bbcc8d2a744c502055e8b/assets/js/modules/analytics-4/datastore/audience-settings.js#L257).
    - [x] `closePanel` prop should be a callback to set `AUDIENCES_SELECTION_PANEL_OPENED_KEY` value to `false`. This callback will also be passed to header component to be triggered on close button's click added in header.
    - [x] Within the  `SelectionPanel` component. Following components needs to be rendered, which would be passed as `children` to the `SelectionPanel` component.
      * `Header`
      * `AudienceItems`
      * `Footer`
  - [x] Create `Header` component inside `AudiencesSelectionPanel` directory.
    - [x] It should make use of `SelectionPanelHeader` and pass following props to it.  
      - [x] `title` - Heading text for the selection panel.
      - [x] `onCloseClick` - Callback to trigger when selection panel is closed. This can be same callback passed to `SelectionPanel`.
      - [x] `children` This should display the text under the main heading with the settings link. This is a component created with `createInterpolateElement`. Refer the [figma design](https://www.figma.com/file/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?type=design&node-id=1106-16323&mode=design&t=h8kDknnPnT4WeS3M-0) for the copy.
  - [ ] Create `AudienceItems` component inside `AudiencesSelectionPanel` directory.
    - [x] It should leverage the `SelectionPanelItems` component and pass the necessary props to it.
    - [x] Pass current selection title and available selection title according to figma designs.
    - [x] `availableSavedItems` and `availableUnsavedItems` should be calculated by fetching available audiences using `getAvailableAudiences` selector which returns the available audiences in property and `getConfiguredAudiences` which returns the saved audiences in DB.
    - [x] `getAvailableAudiences` should be modified to check if "Purchasers" audience has had any data. This can be checked with `getResourceDataAvailabilityDate` from the [partial data](https://github.com/google/site-kit-wp/blob/167c32309e253fdde7e424f9fb05f199ce4ee215/assets/js/modules/analytics-4/datastore/partial-data.js#L205-L337) infra. If there is any data available for the "Purchasers" i.e. `getResourceDataAvailabilityDate` does not return 0 or `undefined`, we should keep "Purchasers" in `getAvailableAudiences` response.
    - [x] If there has never been any data for "Purchasers", filter out the "Purchasers" from `getAvailableAudiences` response as we do not want to show "Purchasers" audience in this case.
    - [ ] While passing the `availableSavedItems` and `availableUnsavedItems` props, they must be mapped to include `title`, `description`, `subtitle` and `userCount` properties. 
      - [x] `title` would be `displayName` from audiences response.
      - [ ] `description` belongs to `description` property.
      - [ ] `subtitle` is the text which appears below description. As seen in the [figma designs](https://www.figma.com/design/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?node-id=1106-16323&t=nDorPVu55GxzZpJz-0), it will be `Created by ....` text. 
        - [ ] If the audience is "All users" or "Purchasers", `subtitle` would be "Created by default by Google Analytics"
        - [ ] If the audience is "New visitors" or "Returning visitors", `subtitle` would be "Created by Site Kit".
        - [ ] For all other audience types, `subtitle` should be "Already exists in your Analytics property".
      - [x] `userCount` would be the total number of users in audience for the selected period. This should use the `getReport` selector from analytics module where the metric would be `totalUsers` and dimension is `audienceResourceName`. 
      ```js
      const options = {
        startDate: "2024-05-09",
        endDate: "2024-05-15",
        metrics: "totalUsers",
        dimensions: [ 'audienceResourceName' ]
      };

      const result = googlesitekit.data.select('modules/analytics-4').getReport( options );
      ```
    - [ ] Purchasers audience should only be listed if it’s had any users in it since the associated Analytics property was created. For the same, we can use the `isResourcePartialData` [selector](https://github.com/google/site-kit-wp/blob/6988aa0b013216146dea2dd267c96734191418e4/assets/js/modules/analytics-4/datastore/partial-data.js#L390-L435) to check if there "Purchasers" has had any audience.
    - [x] For `savedItemSlugs` pass the value returned by `getConfiguredAudiences` which is array of `name` in audience [response](https://github.com/google/site-kit-wp/blob/c9c32590cf0ae896de2c90577dcfcef9a0c2d542/assets/js/modules/analytics-4/datastore/__fixtures__/available-audiences.json#L1-L38).
    - [x] `ItemComponent` should be `AudiencesItem` which will be responsible to render individual selection item in the panel.
      - [x] This component will receive `title`, `description`, `subtitle` and `userCount` props as mentioned above.
      - [x] Note that we need to pass `userCount` as `suffix` into (maybe wrapped into some JSX) `SelectionPanelItem` component. `suffix` prop is described in below point.
  - [x] Add an additional prop `suffix` to the component `SelectionPanelItem`. This should be a react component which should be rendered after the [SelectionBox](https://github.com/google/site-kit-wp/blob/167c32309e253fdde7e424f9fb05f199ce4ee215/assets/js/components/SelectionPanel/SelectionPanelItem.js#L41-L51) component.
    - [ ] Default value of `suffix` can be a function which returns `false` ( `suffix = () => false` ), so that if nothing is passed, it should not render anything. This is particularly useful in case of KM panel.
  - [x] Create `AudiencesItem` component inside `AudiencesSelectionPanel` directory. 
    - [x] It should leverage the `SelectionPanelItem` component and pass the necessary props to it. All these props should already be passed by `SelectionPanelItems` component down to this component.
    - [ ] We can pass `description` as react component which encompasses of `description` of the audience and `created by` subtitle.
  - [x] Create `Footer` component inside `AudiencesSelectionPanel` directory.
    - [ ] This component should use `SelectionPanelFooter` component to return the new composed component for audience segmentation selection panel footer.
    - [x] `savedItemSlugs` - These are the items returned from `getConfiguredAudiences`.
    - [x] `selectedItemSlugs` - This can be retrieved from `AUDIENCES_SELECTED` inside `AUDIENCES_SELECTION_FORM`. Please refer this example [here](https://github.com/google/site-kit-wp/blob/c9c32590cf0ae896de2c90577dcfcef9a0c2d542/assets/js/components/KeyMetrics/MetricsSelectionPanel/MetricsFooter.js#L67-L72).
    - [x] Refer other props in `SelectionPanelFooter` and pass them accordingly. A [reference](https://github.com/google/site-kit-wp/blob/c9c32590cf0ae896de2c90577dcfcef9a0c2d542/assets/js/components/KeyMetrics/MetricsSelectionPanel/MetricsFooter.js#L1-L247) of `MetricsFooter` component can be taken to build other props.
  - [ ] It's worth noting that generic components extracted from the KM Selection Panel should be reused in audience selection panel. Note that footer changes in audience selection panel should also be applied to the KM panel.
    - [ ] In `assets/sass/components/global/_googlesitekit-selection-panel.scss`, target `.googlesitekit-selection-panel-footer__content` and make it display block.
    - [ ] `.googlesitekit-selection-panel-footer__item-count` should be text aligned left and `.googlesitekit-selection-panel-footer__actions` should be text aligned right with display block.
    - [ ] It's essential that both audience selection panel and KM panel should reflect this change.

### Test Coverage

- Add the `AudiencesSelectionPanel` in storybook with the following scenarios.
  - `Default` - Where it should show the flat list of all the audiences.
  - `SavedSelection` - Where it should show the `Current selection` and `Additional groups`. `Current selection` will show the saved items.
  - `View-only user` - With view only context.

- Write the test for `AudienceSegmentation/AudienceSelectionPanel` component. Take reference of KM component [tests](https://github.com/google/site-kit-wp/blob/develop/assets/js/components/KeyMetrics/MetricsSelectionPanel/index.test.js). Not all of the tests in the `MetricsSelectionPanel` suite will apply to the `AudienceSelectionPanel`, so copy/modify the tests which are applicable and add any additional tests as needed.

## QA Brief

* Open Storybook.
  * Open the Modules -> Analytics4 -> Components -> AudienceSegmentation -> Dashboard -> AudienceSelectionPanel` stories.
  * Verify the audience selection panel looks and behaves according to the Figma designs and the ACs.

## Changelog entry

* Add the Audience Selection Panel as a component which is primarily visible in Storybook pending full integration.

**Comment by techanvil** [comment]
Hey @hussain-t, just a heads up that I've amended the Feature Description here to make it clear that we do want to include the logic described in **available audiences** as part of this issue.

I've also tweaked that section in the design doc to point out that the implementation can leverage the "collecting data" infrastructure, and as a result I've added https://github.com/google/site-kit-wp/issues/8141 to the dependencies for this issue.

**Comment by techanvil** [comment]
Hi @hussain-t, just a heads up that I've made a small tweak to the AC to make it clear that we should be leveraging the [audience cache](https://github.com/google/site-kit-wp/issues/8486) here, as opposed to retrieving the audiences directly from the audiences API.

**Comment by techanvil** [comment]
Hi @hussain-t, I've also added a point to the AC to make it clear we don't need to address the specific details of audience ordering in this issue, this will be handled via #8519. This shouldn't actually have any impact on this issue, just want to give a heads up for clarity.

**Comment by hussain-t** [comment]
@ankitrox, could you set an estimate?

**Comment by ankitrox** [comment]
@hussain-t Added the estimate. 
Thank you.

**Comment by techanvil** [comment]
Hi @ankitrox, thanks for drafting this IB. Here are some points to consider:

- Using the single `SELECTION_PANEL_OPENED_KEY` key to control the panel's open state is likely to cause problems when there are more than one version of the panel. I can imagine the shared `SELECTION_PANEL_HEADING` and `SELECTION_PANEL_SELECTED_ITEMS` potentially being problematic too. I think we should keep separate state for each version of the panel. This could be achieved by passing different key values into the component via props, or potentially via context to facilitate their use in child components. We might want to simplify things a bit by keeping all these values within their own form slice, so we effectively just need to pass a unique value for what's currently `SELECTION_PANEL_FORM`.
- Additionally the `MIN_SELECTED_COUNT` and `MAX_SELECTED_COUNT` values will differ for the audiences selection panel which has a min of 1 and max 3 selected items, so these values will also need to be passed via props/context.
- The passing of a `children` property in `selectedItems` and `availableItems` doesn't look right. Rather we should pass a prop like `ItemComponent={Label}` which would be the component function rather than a JSX rendered instance. Then we can render the component for each item, passing the item to it via props e.g. `<ItemComponent item={item} />`.
- Additionally, the shape of the selectable items will be different for each instance - i.e. [Key Metrics objects](https://github.com/google/site-kit-wp/blob/661515c3d39e861d55dc1983fa86daaf2574df73/assets/js/components/KeyMetrics/key-metrics-widgets.js#L125) have a different shape to [cached audience objects](https://github.com/google/site-kit-wp/blob/661515c3d39e861d55dc1983fa86daaf2574df73/assets/js/modules/analytics-4/datastore/__fixtures__/available-audiences.json). We'll probably want to map these to a common shape so we'd want to effectively define an interface for a selectable item and map the specific items to it. I also think `selectedItems` should be a simple array of slugs, while `availableItems` should most likely be an array of mapped, common objects. I imagine these objects should have a shape along the lines of `{ slug, title, subtitle, description }` (considering the commonalities between the KM selection panel and the audiences selection panel):
  - KM panel:  
![image](https://github.com/google/site-kit-wp/assets/18395600/28d85d53-356d-4d9f-97ad-358c87f49e10)
  - Audiences panel:  
![image](https://github.com/google/site-kit-wp/assets/18395600/e28aa257-2242-4e1c-a094-4f4a30169f39)
- In fact, taking these commonalities into consideration maybe we don't even need `ItemComponent` and it should rather be an optional component for rendering a "suffix", e.g. `ItemSuffixComponent`, or `ItemDetailComponent` or something, which can then be used just to show the user count for the audiences panel.
- There are a number of Key Metric specific aspects that have been glossed over/missed and will need extracting/abstracting from the appropriate components, for example the [saving state](https://github.com/google/site-kit-wp/blob/fc0460bcaf391648321dc5eebc8389edc785a77d/assets/js/components/KeyMetrics/MetricsSelectionPanel/Header.js#L45-L47), [subheader text](https://github.com/google/site-kit-wp/blob/fc0460bcaf391648321dc5eebc8389edc785a77d/assets/js/components/KeyMetrics/MetricsSelectionPanel/Header.js#L77) and [behaviour](https://github.com/google/site-kit-wp/blob/fc0460bcaf391648321dc5eebc8389edc785a77d/assets/js/components/KeyMetrics/MetricsSelectionPanel/Header.js#L56-L59) in the `Header` component are all KM-specific, and there are similar things across all the child components. It's worth considering that by lifting this logic out of the child components we might find ourselves with an overly complex parent so it could be appropriate to extract some reusable "base" child components which we then create instance-specific wrappers in which can encapsulate the logic for those components. E.g. we extract a common `Header` but also create, say, a `MetricsSelectionPanel/Header` which contains the KM-specific logic for the header and renders the common `Header` within it. We don't necessarily need to get too detailed about these aspects in the IB, and we could make a general point about them for awareness rather than defining each and every prop.
- CSS classnames should be renamed to remove/replace the `km` substrings so as to be generic. We should avoid passing `className` in unless we actually need it, a key thing we want to get out of this refactor is to keep the styling common as much as possible.
- The IB implies that we'd need to wait for #8158 to be merged in order to retrieve the list of selected widgets. However, #8158 is a dependant of this issue so this issue needs to be merged first. We can simply specify that the list of selected widgets can be retrieved via the `configuredAudiences` settings for the purpose of this issue.
- There are some minor formatting issues where backticks haven't been applied to variable names, for example `selectedItems` at the start of point A.3.i and similar. Also it looks like the code snippet in point C.3.i should be using the triple-backtick formatting. There's also a typo that has crept into the IB and POC, "metricss", should obviously be "metrics".
- As a general point it's worth bearing in mind this is quite a complex issue for you to have picked up for IB quite early into your time on the project. It's commendable to be keen to get stuck in, but it might be worth ramping up a bit more slowly, going for some easier issues and building up progressively, if possible. I know we are a bit limited in terms of what's available to be picked up so it might be difficult to apply this so well in practice.

**Comment by techanvil** [comment]
Note that since my previous comment we've decided to split this issue into two, with https://github.com/google/site-kit-wp/issues/8652 being created for the initial KM Selection Panel refactor.

**Comment by ankitrox** [comment]
Thank you @techanvil for breaking this task into refactoring and audience segmentation panel. This is helpful to better handle the overall requirement.

I have updated the IB considering the fact that we have a separate task for refactoring and key metrics panel. This IB take reference of your POC PR, but I think that should be fine because we are composing the components using extracted components.

Assigning this to you for review.

Thanks again!

**Comment by techanvil** [comment]
Hey @ankitrox, thanks for updating the IB. However - you've accidentally removed the AC and replaced the AC section with the new IB. Please can you restore the AC (you can see this in the edit history) and send it back to me to continue the review?

**Comment by ankitrox** [comment]
@techanvil Extremely sorry for the inconvenience! 
I have restored the AC from issue's history. 

Assigning it over to you to continue the review.

Thanks

**Comment by techanvil** [comment]
Hi @ankitrox, no worries! These things happen, luckily we have the edit history to help us out here. Thanks for restoring the AC. As to the IB, it's generally looking good - here are a few points to address:

- I'd suggest we name the top-level component `AudiencesSelectionPanel` which would be more in keeping with the `MetricsSelectionPanel` naming, with corresponding renames from `AudienceSegmentationHeader` to `AudiencesHeader`, etc.

---

> If there are no saved audiences, we should always display user created audiences which has non-zero user count, under "Current Selection". User count will be fetched by `getReport` selector in analytics module.

- This point is not accurate. We should _only_ show saved audiences in the "Current selection" grouping. If there are no saved audiences, the selection panel will be displayed as a flat list, similar to the Key Metrics selection panel when there are no saved metrics, as can be seen here in [Storybook](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/key-metrics-metricsselectionpanel--default). We don't need to filter user created audiences to those with a non-zero user count within the selection panel.

---

> "All visitors" (which are "All users" in analytics and renamed in site kit), will be shown under "Additional groups" if not selected.

- The phrasing "will be shown under "Additional groups" if not selected" is a little confusing. In fact, the "All visitors" audience, like any other audience, will be shown under "Current selection" if it's part of the _saved_ selection, otherwise it will be displayed under "Additional groups" (or as part of the flat list if there is no saved selection).
- I'd suggest this point could be tweaked along these lines:
  - "All visitors" (which are "All users" in analytics and renamed in Site Kit), will be initially shown under "Additional groups" (assuming there's a saved selection, otherwise it will be part of the flat, ungrouped list of available audiences).

---

> "Purchasers" will be shown in "current selection" if and only if it has user count for past 90 days and has not been archived.

- This needs a correction, and it's worth expanding a bit on the implementation here.
  - Again, there's a bit of confusion as a result of explicitly specifying the "current selection". As discussed above the audience will be displayed in "Current selection", "Additional groups" or the flat list depending on whether it's part of the saved selection (and if there _is_ a saved selection). I'd suggest a similar tweak as the one for "All visitors" above.
  - The AC specifies that the Purchasers audience "should only be listed if it’s had any users in it since the associated Analytics property was created". This is not the same as having a user count over the past 90 days. To check if the audience meets the AC we can make use of the "partial data" data availability state, as this involves caching the earliest event for an audience. So we can make the "partial data" check for the audience and if there _is_ an earliest event, we know the audience has had some data since the property was created.
  - Note that if an audience has been archived, it won't appear in `availableAudiences`.

---

>  It should leverage the `SelectionPanelItem` component and pass the necessary props to it. All these props should already be passed by `SelectionPanelItems` component down to this component.

- We probably need to extend `SelectionPanelItem` to include a `subtitle` prop (as per the related point in this [comment](https://github.com/google/site-kit-wp/issues/8157#issuecomment-2088747787)), and also some way of passing/rendering the user count (e.g. a "suffix" as we have discussed previously) - or, maybe the user count can be rendered directly via `AudiencesItem`. It's worth a mention, either way.

---

>  Create `AudienceCreationNotice` component which will be displayed only if site kit created audiences i.e. `New visitors` and `Returning visitors` are not available in audiences returned by `getAvailableAudiences` selector.

- We don't actually need to create this component as part of this issue. It can be done as part of the linked issue #8164.

---

> Add the AudienceSegmentationPanel in storybook with the following scenarios.

- This point in the Test Coverage section looks incomplete as the scenarios are not specified.

---

- Regarding the following AC point:

> * The generic components extracted from the KM Selection Panel should be reused (see google/site-kit-wp#8652), with changes to the design for Audience Segmentation applied back to the KM panel (notably the footer layout in the mobile view as mentioned in the [design doc](https://docs.google.com/document/d/1MGD5Djy6AeeZC4zBtHqS-lQEWD9jw0kf-IIWw-jLCFU/edit#heading=h.qt81tly4no4r)).

- ... it's worth a mention of this somewhere in the IB as it could be easy to miss otherwise.

**Comment by techanvil** [comment]
Hey @ankitrox, thanks for iterating on this IB. A few further points:

- `AUDIENCE_SELECTION_SETUP_CTA_WIDGET_SLUG` is only referenced once in the IB and it's not clear we'll actually need it at this stage. This should be removed.
- `AUDIENCE_SELECTION_PANEL_OPENED_KEY`, `AUDIENCE_SELECTION_FORM` should be renamed for consistency with KM plurality, see https://github.com/google/site-kit-wp/blob/cbc96d3c9a51af7a42a7d4d779818b36a95cd64f/assets/js/components/KeyMetrics/MetricsSelectionPanel/index.js#L45. So they should be `AUDIENCES_SELECTION_PANEL_OPENED_KEY` and `AUDIENCES_SELECTION_FORM`.
- `closeFn()`, `AudiencesHeader`, `AudiencesItems`, `AudiencesFooter` need renaming due to changes in the implementation for #8652, (see its [PR](https://github.com/google/site-kit-wp/pull/8698/files)), they should be renamed to `closePanel()`, `Header`, `AudienceItems` and `Footer` respectively.

---

- The following points (I've just included the start of each of them for brevity) read more like AC for the most part but don't really need including here. They can be removed as from the implementation POV we just need to populate via `getAvailableAudiences()` but remove Purchasers if it's not had any data. As mentioned these points can be removed but the point about calculating `availableUnsavedItems` should be expanded on.

> - If there are no saved audiences...
> - All visitors" (which are "All users" in analytics and renamed in Site Kit), will be initially shown...
> - "Purchasers" will also be initially shown under "Additional groups"...

- With regard to the Purchasers audience availability, I realise I was slightly unclear in my previous feedback. We should not use the state returned by `isResourcePartialData()`, but in fact the underlying state returned by `getResourceDataAvailabilityDate()`. This is because `isResourcePartialData()` relates to the current date range, but we want to know if there has _ever_ been any data for the audience.

---

> - Note that we need to change `SelectionPanelItems` component a bit so that we can pass `children` to `ItemComponent`.
> - `children` passed to the `ItemComponent` can be a react component useful to display the user count for the audience.

- I don't see we need to do this. For one thing, I still think we should allow `SelectionPanelItem` to be passed e.g. a `suffix` which can be rendered within `.googlesitekit-selection-panel-item` but after the `SelectionBox`. Then, bearing in mind that each item is spread into `ItemComponent` as props:

https://github.com/google/site-kit-wp/blob/f6ea234136d2ca2e70aaf166e6694c2d0bfbed59/assets/js/components/SelectionPanel/SelectionPanelItems.js#L40-L45

- ... we don't need to update `SelectionPanelItems`, we can simply ensure that each item has say a `subtitle` and `userCount` property, update the props for `AudienceItem` to be `{ slug, title, description, subtitle, userCount, savedItemSlugs = [] }`, and render  `SelectionPanelItem` within it accordingly with `description` and `subtitle` folded into `description` and `userCount` passed as `suffix` (maybe wrapped in a bit of extra JSX).
- Further to the above, the IB hasn't mentioned retrieving the user counts. We don't need to go into too much detail, but essentially we need to make a report for the user counts for all audiences (so retrieve the `totalUsers` metric by the `audienceResourceName` dimension for the current date range), and then use the report to populate the `userCount` property for the audience items.


---

> It's worth noting that generic components extracted from the KM Selection Panel should be reused in audience selection panel. We may also need to apply some design changes; for example audience selection panel footer is slightly different than the KM selection panel footer (in mobile view). In such cases, we can target the particular elements by passing appropriate class names and styling them differently in CSS. We may also need slightly different components to be passed, for example; in `SelectionItem` we may need different components for `description` and `children`.

- This is not quite right, because as per the last section in my previous review, we actually want to ensure that the changes to the footer styling for the Audiences panel are applied back to the generic footer component and thus the KM panel too, ensuring we keep a consistent look and feel to both panels.

---

Please let me know if it would be useful to jump on a call to discuss/clarify any of the above, I realise it's quite a complicated one that might benefit from a sync.

**Comment by techanvil** [comment]
Thanks for updating the IB, @ankitrox. It's LGTM (please note I made a small tweak to the Test Coverage section which I felt could be a little less explicit).

IB :white_check_mark:

**Comment by techanvil** [comment]
Hi @nfmohit, nice work here. I've just merged the PR.

However - considering this issue was only intended to be scoped to Storybook, I am questioning if we need to include the integration testing in the QAB. I note that you've rendered `AudienceSelectionPanel` in `DashboardMainApp` which I can imagine was useful for additional testing during development, and I've left that in there, but really it's a bit out of scope here. Considering the issue has gone over the estimate I'd suggest removing the E2E tests from the QAB and instead use those as a basis for the QAB for #8158 which I see you currently have in execution. WDYT?

**Comment by nfmohit** [comment]
> Hi @nfmohit, nice work here. I've just merged the PR.
> 
> However - considering this issue was only intended to be scoped to Storybook, I am questioning if we need to include the integration testing in the QAB. I note that you've rendered `AudienceSelectionPanel` in `DashboardMainApp` which I can imagine was useful for additional testing during development, and I've left that in, but really it's a bit out of scope here. Considering the issue has gone over the estimate I'd suggest removing the E2E tests from the QAB and instead use those as a basis for the QAB for #8158 which I see you currently have in execution. WDYT?

That's a good call, thank you @techanvil. I've moved out all the plugin testing from the QAB and just left the parts for Storybook. Let me know what you think, thank you!

**Comment by techanvil** [comment]
Thanks @nfmohit, that looks great. Over to QA for testing :)

**Comment by kelvinballoo** [comment]
### QA Update ⚠️ 

Tested and I have a few observations and queries, documented as follows:

**ITEM 1:**
All visitors' should be 'Google Sans Text' instead of 'Google Sans Display'

<Details>
Storybook:
![Uploading Screenshot 2024-06-06 at 16.11.57.png…]()

Figma:
<img width="973" alt="Screenshot 2024-06-06 at 16 12 54" src="https://github.com/google/site-kit-wp/assets/125576926/7994995b-5479-4971-8f7a-0750d752f051">

</Details>

**ITEM 2:**
**People who visited the site for the first time** font weight should be 400 instead of 500.

<Details>
Storybook:
<img width="714" alt="Screenshot 2024-06-06 at 16 14 20" src="https://github.com/google/site-kit-wp/assets/125576926/874f736e-a2d7-4569-976a-72d8d09823ca">

Figma:
<img width="924" alt="Screenshot 2024-06-06 at 16 15 31" src="https://github.com/google/site-kit-wp/assets/125576926/ceaf93c5-a95b-4a6e-913b-cbf37902ae7b">
</Details>

**ITEM 3:**
Font weight for 'Save selection' button should be 500 instead of 400.
That said, I did see that the KM panel would be 400. So if this is to keep things consistent with KM, then we can ignore this.

<Details>
Storybook:

<img width="380" alt="Screenshot 2024-06-06 at 16 17 41" src="https://github.com/google/site-kit-wp/assets/125576926/0961becf-e072-4fc3-ae10-9edfb3a9df66">

Figma:

<img width="668" alt="Screenshot 2024-06-06 at 16 17 57" src="https://github.com/google/site-kit-wp/assets/125576926/aa18afaf-cb98-484d-8fe3-e294478e8570">
</Details>


**ITEM 4:**
There are a few points in the AC that I don't think can be tested in this Storybook ticket as the storybook data are not an data linked to an real account. I've listed them  below.
Can you confirm that indeed these are items not to be tested in this ticket 

- The Selection Panel should display the list of available audiences that is cached in the DB (i.e. stored in the availableAudiences setting). Their display names and descriptions should be displayed directly, with the exception of the "All users" default audience which should be shown as "All visitors".

- The “Purchasers” default audience should only be listed if it’s had any users in it since the associated Analytics property was created.
- The user count for an audience should be displayed on the right hand side of it.
- The Selection Panel should display the header text including the link to the Admin Settings tab on the Settings screen.
   - If the Audience Segmentation Settings section is available at implementation time this link should ensure the section is scrolled into view when navigating to the Settings screen. Otherwise, a small followup issue should be created to update the link when the section is available."

**Comment by nfmohit** [comment]
Thank you for sharing your observations, @kelvinballoo !

> **ITEM 1:** All visitors' should be 'Google Sans Text' instead of 'Google Sans Display'
> 
> Storybook: ![Uploading Screenshot 2024-06-06 at 16.11.57.png…]()
> Figma: <img alt="Screenshot 2024-06-06 at 16 12 54" width="973" src="https://private-user-images.githubusercontent.com/125576926/337258399-7994995b-5479-4971-8f7a-0750d752f051.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2Nzg1MTAsIm5iZiI6MTcxNzY3ODIxMCwicGF0aCI6Ii8xMjU1NzY5MjYvMzM3MjU4Mzk5LTc5OTQ5OTViLTU0NzktNDk3MS04ZjdhLTA3NTBkNzUyZjA1MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwNlQxMjUwMTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MzIzMWVlMjNlYmMwNmU1MmI1OTAwZmI4NDE0OTY1MjM5NmY1NWE1MDZhZGU3OGE1MTc3ZGFiODYxYTgzZTE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.N9fjlDZgXCfRES-Qlz5JAeY5heLhBvpxYlfGSwcz3tQ">

This is consistent with what we have in the Key Metrics selection panel. If we should change the font here, I believe we should change it in the KM panel as well for consistency. @sigal-teller Could you confirm if the font should be changed here to `Google Sans Text` for both KM and audience selection panels?

> **ITEM 2:**
> **People who visited the site for the first time** font weight should be 400 instead of 500.

I'm opening a follow-up PR for this.

> **ITEM 3:**
> Font weight for 'Save selection' button should be 500 instead of 400.
> That said, I did see that the KM panel would be 400. So if this is to keep things consistent with KM, then we can ignore this.

As you mentioned, this uses the re-usable component that both KM and audience selection panels use, so if we want to change it here, we should change it everywhere else for consistency.. @sigal-teller We need your confirmation here as well to see if this change should be applied for all selection panels. Thanks!

> * The Selection Panel should display the list of available audiences that is cached in the DB (i.e. stored in the availableAudiences setting). Their display names and descriptions should be displayed directly, with the exception of the "All users" default audience which should be shown as "All visitors".
> * The “Purchasers” default audience should only be listed if it’s had any users in it since the associated Analytics property was created.

As this is a Storybook-only issue, unfortunately, these cannot be tested here, but can be done in #8158. I'll add them in the QAB for #8158.

> * The user count for an audience should be displayed on the right hand side of it.

You should be able to see the user count from the mock report data in Storybook.

![image](https://github.com/google/site-kit-wp/assets/20284937/5f3d62db-01f3-44c5-9328-14b692e86234)

You'll be able to see actual data in #8158.

> The Selection Panel should display the header text including the link to the Admin Settings tab on the Settings screen.
> 
> * If the Audience Segmentation Settings section is available at implementation time this link should ensure the section is scrolled into view when navigating to the Settings screen. Otherwise, a small followup issue should be created to update the link when the section is available."

The link is correct, however, unfortunately, you'll not be able to see it in Storybook. It should be verifiable in #8158.

The link doesn't scroll down to the respective section though. @techanvil [mentioned](https://github.com/google/site-kit-wp/pull/8751#discussion_r1626352985) he is going to open an issue for it.

Thanks!

**Comment by nfmohit** [comment]
I have opened a follow-up PR #8829 to address some of the concerns above. For this to move forward, we have two questions for @sigal-teller that we need for confirmation:

1.
> > **ITEM 1:** All visitors' should be 'Google Sans Text' instead of 'Google Sans Display'
> > 
> > Storybook: ![Uploading Screenshot 2024-06-06 at 16.11.57.png…]()
> > Figma: <img alt="Screenshot 2024-06-06 at 16 12 54" width="973" src="https://private-user-images.githubusercontent.com/125576926/337258399-7994995b-5479-4971-8f7a-0750d752f051.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2Nzg1MTAsIm5iZiI6MTcxNzY3ODIxMCwicGF0aCI6Ii8xMjU1NzY5MjYvMzM3MjU4Mzk5LTc5OTQ5OTViLTU0NzktNDk3MS04ZjdhLTA3NTBkNzUyZjA1MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwNlQxMjUwMTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MzIzMWVlMjNlYmMwNmU1MmI1OTAwZmI4NDE0OTY1MjM5NmY1NWE1MDZhZGU3OGE1MTc3ZGFiODYxYTgzZTE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.N9fjlDZgXCfRES-Qlz5JAeY5heLhBvpxYlfGSwcz3tQ">
> 
> This is consistent with what we have in the Key Metrics selection panel. If we should change the font here, I believe we should change it in the KM panel as well for consistency. @sigal-teller Could you confirm if the font should be changed here to `Google Sans Text` for both KM and audience selection panels?

2.
> > **ITEM 3:**
> > Font weight for 'Save selection' button should be 500 instead of 400.
> > That said, I did see that the KM panel would be 400. So if this is to keep things consistent with KM, then we can ignore this.
> 
> As you mentioned, this uses the re-usable component that both KM and audience selection panels use, so if we want to change it here, we should change it everywhere else for consistency.. @sigal-teller We need your confirmation here as well to see if this change should be applied for all selection panels. Thanks!

Thank you!

**Comment by sigal-teller** [comment]
@nfmohit The links you added appears broken for me so I can see exactly what you're referring to. In any case, I'm using the styles that we have in the design system, so I'm not sure why the font is changing for you. Maybe there are missing fonts or styles for you and this is why replaces them? Anyway, group names (if that was what you were referring to) are label medium. LMK if you were referring to something else.

**Comment by nfmohit** [comment]
Hi @sigal-teller. Could you check out [this link](https://www.figma.com/design/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?node-id=1398-26857&t=0Ou6D9SZY0vERGB5-4)? Let me describe further below:

In Figma, the font for "Change Metrics" is "Google Sans Text". Screenshot:

![image](https://github.com/google/site-kit-wp/assets/20284937/c7b97dc6-4e13-474a-b4da-2bdb5f56168a)

But in Site Kit, it is "Google Sans Display", most likely it was "Google Sans Display" in Figma too when that was implemented. See screenshot:

![image](https://github.com/google/site-kit-wp/assets/20284937/bba6f17a-8831-4b81-8aae-9e01431b5526)

My question is, should we update this to be "Google Sans Text" according to the current designs? Thanks!

**Comment by nfmohit** [comment]
Hi @sigal-teller. I have updated the links for you, thank you for pointing that out. In summary, we have two questions:

1st:
Should we use "Google Sans Text" for the checkbox labels in the selection panel? The KM panel, when implemented, used the "Google Sans Display" font, but in the current Figma designs for Audience Segmentation, it uses "Google Sans Text".

<details>
<summary>Screenshots</summary>

In the current KM selection panel:
![image](https://github.com/google/site-kit-wp/assets/20284937/ca420901-f529-483c-83ef-152c3728fe26)

In Audience Segmentation Figma designs:
![image](https://github.com/google/site-kit-wp/assets/20284937/9aae9edc-2448-43e2-99fe-f1bcee603d3d)
</details>

2nd:
Should the font-weight for the "Save selection" CTA be 400 or 500. In the KM selection panel, it is 400, but in the Audience Segmentation Figma designs, it is 500.

<details>
<summary>Screenshots</summary>

In the current KM selection panel:
![image](https://github.com/google/site-kit-wp/assets/20284937/a6ea0924-10c4-4960-8b49-330c03581868)

In the Audience Segmentation Figma designs:
![image](https://github.com/google/site-kit-wp/assets/20284937/333d7310-fc89-480e-855d-a7a49c403f87)

</details>

**Comment by sigal-teller** [comment]
Hi @nfmohit
1. Checkboxes labels should be "Google sans text" (this what is used for Label/medium" in Figma).
2. font weight for buttons should be 500 (buttons text are Label/medium as well). you can find them in the design system [here](https://www.figma.com/design/ZL8qlbUyR9lmBaF95t0xpq/sitekit-system-gm2%2B?node-id=1608-9409&t=qUYdfWfzy9M1VVfk-1). Also attaching a screenshot.
<img width="1096" alt="Screenshot 2024-06-11 at 16 57 35" src="https://github.com/google/site-kit-wp/assets/69191559/11e89583-07ea-4760-a358-b9b1592df392">

**Comment by nfmohit** [comment]
Thank you for confirming, @sigal-teller. I'll make these updates accordingly.

**Comment by nfmohit** [comment]
@kelvinballoo This is now in CR with PR #8829 that addresses your observations: **ITEM 1** & **ITEM 2**.

For **ITEM 3**, I have opened a new issue #8856.

The other questions have been addressed in my previous comments.

Thanks!

**Comment by kuasha420** [comment]
@kelvinballoo Back to you for another pass, the ITEM 1 and 2 should now be fixed. As noted by @nfmohit above, the 3rd item is a much bigger change, so it will be fixed in a follow up issue.

**Comment by kelvinballoo** [comment]
### QA Update ✅ 

All the issues are resolved or have an action in another ticket. Moving this ticket to 'Approval'.

**ITEM 1:** ✅ 
All visitors' is now 'Google Sans Text'.
<Details>
<img width="912" alt="Screenshot 2024-06-12 at 17 20 20" src="https://github.com/google/site-kit-wp/assets/125576926/50a7addd-89e9-4493-9941-e1760bf77924">

</Details>
________________________________________________________________

**ITEM 2:** ✅ 
**People who visited the site for the first time** font weight is now 400.
<img width="918" alt="Screenshot 2024-06-12 at 17 22 12" src="https://github.com/google/site-kit-wp/assets/125576926/4255b86b-a3e5-424f-bbd4-f2bd7286b89a">

________________________________________________________________

**ITEM 3:** ✅ 
Noted that this will be handled under https://github.com/google/site-kit-wp/issues/8856.
________________________________________________________________

**ITEM 4:** ✅ 
Noted that this is out of scope for this test.
